src/javascript/app_2/App/Containers/AccountSwitcher/account-switcher.jsx

```diff
@@ -88,22 +91,25 @@ class AccountSwitcher extends React.Component {
                         }
                     </div>
                 }
-                <div className='acc-switcher__list--virtual'>
-                    <span className='acc-switcher__list-title'>
-                        {localize('Virtual account')}
-                    </span>
-                    <div
-                        className={classNames('acc-switcher__account', {
-                            'acc-switcher__account--selected': (this.props.virtual_loginid === this.props.account_loginid),
-                        })}
-                        onClick={this.doSwitch.bind(this, this.props.virtual_loginid)}
-                    >
-                        <span className={classNames('acc-switcher__id', 'acc-switcher__id--virtual')}>
-                            {this.props.virtual_loginid}
+                {
+                    !isEmptyObject(vrt_account) &&
+                    <div className='acc-switcher__list--virtual'>
+                        <span className='acc-switcher__list-title'>
+                            {localize('Virtual account')}
                         </span>
-                        <span className='acc-switcher__radio' />
+                        <div
+                            className={classNames('acc-switcher__account', {
+                                'acc-switcher__account--selected': (this.props.virtual_loginid === this.props.account_loginid),
+                            })}
+                            onClick={this.doSwitch.bind(this, this.props.virtual_loginid)}
```

Comment:
why not binding this to doSwitch in constructor method?  to prevent react re-render (arrow functions and inline bindings make the component re-render every time)

Reply:
@nazaninreihani that's for virtual account, right, and so I can't seem to work that out with the other part for Real Accounts as it relies on the `map()` call to map the individual accounts, and the `loginid` of the account is passed to `doSwitch()`. Do you have any suggestions? here's the code:

```
real_accounts.map(account => (
  <div
    key={account.loginid}
    className={classNames("acc-switcher__account", {
      "acc-switcher__account--selected":
        account.loginid === this.props.account_loginid
    })}
    onClick={this.doSwitch.bind(this, account.loginid)}
  >
    <span className={"acc-switcher__id"}>
      <IconAccountsCurrency
        className={`acc-switcher__id-icon acc-switcher__id-icon--${
          account.icon
        }`}
        type={account.icon}
      />
      {account.loginid}
    </span>
  </div>
));
                            
```

Reply:
Okay, I could do all of this before `render()` is called I guess, but for now we're going to create a card for `Optimizing re-rendering components`, so I'll just add this in there too.